flow/src/test/java/com/sequenceiq/flow/helper/FlowStateEnumChecker.java

```diff
@@ -0,0 +1,35 @@
+package com.sequenceiq.flow.helper;
+
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import org.reflections.Reflections;
+import org.reflections.scanners.SubTypesScanner;
+
+import com.sequenceiq.flow.core.FlowState;
+
+public class FlowStateEnumChecker {
+
+    private static final String BASE_PACKAGE = "com.sequenceiq.";
+
+    private final List<String> validationErrors = new ArrayList<>();
+
+    public void checkFlowStateClasses() {
```

Comment:
Not a big deal but this impl is not thread safe. I would rather move `validationErrors` into `checkFlowStateClasses()` as a local var, and pass it to `performCheck()`.

Reply:
it does not matter as stream is processed in a sequential order by default